aptos-move/aptos-vm/src/sharded_block_executor/local_executor_shard.rs

```diff
@@ -0,0 +1,249 @@
+    pub fn new(
+        command_tx: Vec<Sender<ExecutorShardCommand<S>>>,
+        result_rx: Vec<Receiver<Result<Vec<Vec<TransactionOutput>>, VMStatus>>>,
+        executor_shards: Vec<LocalExecutorService<S>>,
+    ) -> Self {
+        Self {
+            command_txs: command_tx,
+            result_rxs: result_rx,
+            executor_services: executor_shards,
+        }
+    }
+}
+
+impl<S: StateView + Sync + Send + 'static> ExecutorClient<S> for LocalExecutorClient<S> {
+    fn num_shards(&self) -> usize {
+        self.command_txs.len()
+    }
+
+    fn execute_block(
+        &self,
+        state_view: Arc<S>,
+        block: Vec<SubBlocksForShard<AnalyzedTransaction>>,
+        concurrency_level_per_shard: usize,
+        maybe_block_gas_limit: Option<u64>,
+    ) {
+        for (i, sub_blocks_for_shard) in block.into_iter().enumerate() {
+            self.command_txs[i]
+                .send(ExecutorShardCommand::ExecuteSubBlocks(
+                    state_view.clone(),
+                    sub_blocks_for_shard,
+                    concurrency_level_per_shard,
+                    maybe_block_gas_limit,
+                ))
+                .unwrap();
+        }
+    }
+
+    fn get_execution_result(&self) -> Result<Vec<Vec<Vec<TransactionOutput>>>, VMStatus> {
+        trace!("LocalExecutorClient Waiting for results");
+        let mut results = vec![];
+        for rx in self.result_rxs.iter() {
+            results.push(rx.recv().unwrap()?);
+        }
+        Ok(results)
+    }
+}
+
+impl<S: StateView + Sync + Send + 'static> Drop for LocalExecutorClient<S> {
+    fn drop(&mut self) {
+        for command_tx in self.command_txs.iter() {
+            let _ = command_tx.send(ExecutorShardCommand::Stop);
+        }
+
+        // wait for join handles to finish
+        for executor_service in self.executor_services.iter_mut() {
+            let _ = executor_service.join_handle.take().unwrap().join();
+        }
+    }
+}
+
+pub struct LocalCoordinatorClient<S> {
+    command_rx: Receiver<ExecutorShardCommand<S>>,
+    // Channel to send execution results to the coordinator.
+    result_tx: Sender<Result<Vec<Vec<TransactionOutput>>, VMStatus>>,
+}
+
+impl<S> LocalCoordinatorClient<S> {
+    pub fn new(
+        command_rx: Receiver<ExecutorShardCommand<S>>,
+        result_tx: Sender<Result<Vec<Vec<TransactionOutput>>, VMStatus>>,
+    ) -> Self {
+        Self {
+            command_rx,
+            result_tx,
+        }
+    }
+}
+
+impl<S: StateView + Sync + Send + 'static> CoordinatorClient<S> for LocalCoordinatorClient<S> {
+    fn receive_execute_command(&self) -> ExecutorShardCommand<S> {
+        self.command_rx.recv().unwrap()
+    }
+
+    fn send_execution_result(&self, result: Result<Vec<Vec<TransactionOutput>>, VMStatus>) {
+        self.result_tx.send(result).unwrap()
+    }
+}
+
+pub struct LocalCrossShardClient {
+    // The senders of cross-shard messages to other shards per round.
```

Comment:
I see that we have the comments here, and it would have helped in the APIs above to have seen this definition. Went back and suggested to pass this struct directly since the parameters are anyway coupled and used to construct LocalCrossShardClient